README.md

```diff
@@ -75,6 +64,65 @@ The unit tests have been implemented by sending a POST request to GitHub and obs
 One test case send valid API data and expects a valid response, another sends invalid API data and
 expects an invalid response.
 
+## API Documentation
+
+### POST requests
+`/push-events`
+
+#### Description
+Pushes an event to the server
+
+#### Parameters
+None
+
+Request body implemented according to [this](https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push)
+
+Request body example:
+```
+{
+    "ref":"refs/heads/mustafa/test-branch",
+    "after":"ea312d0e44d08a603935b8b926eea76e4ed0e266",
+    "repository": {
+        "name":"a2-ci-test",
+        "clone_url":"https://github.com/DD2480-G12/a2-ci-test.git",
+        "owner": {
+            "name":"DD2480-G12"
+        }
+    }
+}
+```
+#### Responses
+204 No Content: This endpoint will always return 204 No Content
+
+### GET requests
+`/history`
+
+#### Description
+Returns the whole stored history of builds
+
+#### Parameters
+None
+
+#### Responses
+200 OK: This endpoint will always return 200 OK. If the history is empty, an empty array will be returned. 
+
+`/history/{id}`
+
+#### Description
+Returns the information for a specific build
+
+#### Parameters
+`id`: Integer, the uid of the desired build
+
+#### Responses
+400 Bad Request: If `id` is not a number\
+404 Not Found: If build `id` does not exist\
+200 OK: If no other response is applicable
```

Comment:
Not sure what you mean by this. It returns 200 if a build log with the given id exists.